api/lib/domain/services/certification-challenges-service.js

```diff
@@ -100,33 +104,51 @@ function _pick3CertificationChallengesForCompetence(
   const result = [];
   const alreadySelectedChallengeIds = _.map(certificationChallengesPickedForOtherCompetences, 'challengeId');
 
-  const orderedSkills = _.orderBy(userCompetence.getSkillsAtLatestVersion(), 'difficulty', 'desc');
-  for (const skill of orderedSkills) {
-    if (_haveEnoughCertificationChallenges(result, MAX_CHALLENGES_PER_COMPETENCE_FOR_CERTIFICATION)) {
-      break;
+  const orderedSkills = _(userCompetence.getSkillsAtLatestVersion())
+    .orderBy('difficulty', 'desc')
+    .groupBy((skill) => skill.difficulty)
+    .value();
```

Comment:
Propositions un peu anecdotiques :
```suggestion
  const orderedSkills = _(userCompetence.getSkillsAtLatestVersion())
    .orderBy('difficulty', 'desc')
    .groupBy('difficulty')
    .value();
```
ou
```suggestion
  const orderedSkills = _.flow(
      fp.orderBy('difficulty', 'desc'),
      fp.groupBy('difficulty')
    )(userCompetence.getSkillsAtLatestVersion());
```

Reply:
J'ai pris la première suggestion.